lib/core/services/log.service.ts

```diff
@@ -15,16 +15,27 @@
  * limitations under the License.
  */
 
+/* tslint:disable:no-console  */
+
 import { Injectable } from '@angular/core';
 import { AppConfigService } from '../app-config/app-config.service';
 import { logLevels, LogLevelsEnum } from '../models/log-levels.model';
+import { Observable } from 'rxjs/Observable';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
 
 @Injectable()
 export class LogService {
 
     currentLogLevel: LogLevelsEnum = LogLevelsEnum.TRACE;
 
+    private onMessageSubject: BehaviorSubject<any>;
+    onMessage: Observable<any>;
+
     constructor(appConfig: AppConfigService) {
+
+        this.onMessageSubject = new BehaviorSubject('');
```

Comment:
If you use Subject<T> you will be able both raising and subscribing with the same object, without 2 different properties.